sound/soc/sof/intel/hda-stream.c

```diff
@@ -590,14 +607,19 @@ static bool hda_dsp_stream_check(struct hdac_bus *bus, u32 status)
 			snd_hdac_stream_writeb(s, SD_STS, sd_status);
 
 			active = true;
-			if (!s->substream ||
+			if ((!s->substream && !s->cstream) ||
 			    !s->running ||
 			    (sd_status & SOF_HDA_CL_DMA_SD_INT_COMPLETE) == 0)
 				continue;
 
 			/* Inform ALSA only in case not do that with IPC */
-			if (sof_hda->no_ipc_position)
+			if (s->substream && sof_hda->no_ipc_position) {
 				snd_sof_pcm_period_elapsed(s->substream);
+			} else if (s->cstream) {
+				hda_dsp_set_bytes_transferred(s,
+					s->cstream->runtime->buffer_size);
```

Comment:
I was misled for a couple of minutes here, I thought the 'set' meant that you'd store the information needed for the next function, but it looks like this is just tracking the rw pointer, so maybe rename this function to e.g. hda_dsp_update_bytes_transferred?

Reply:
Sure, will think of a better function name with 'update' word in mind, thanks!

Reply:
Fixed in v2.